rfc/java17.md

```diff
@@ -0,0 +1,183 @@
+# Java 17 Migration
+
+| | |
+|-|-|
+| **Status**     | Proposed |
+| **RFC #**      | [335](https://github.com/spinnaker/governance/pull/335) |
+| **Author(s)**  | Matt Gogerly ([`@mattgogerly`](https://github.com/mattgogerly)) |
+| **SIG / WG**   | Platform SIG |
+
+## Overview
+
+To keep current with the evolution of Java, Spinnaker needs to move from Java 11
+to Java 17. This RFC documents a process and timeline for doing so.
+
+### Goals and Non-Goals
+
+Goals:
+
+*   Provide a process for moving official Spinnaker builds to the Java 17 JRE
+*   Discuss ways we will encourage people using non-offical builds to also move
+    to the Java 17 JRE
+*   Provide a timeline for the final switch to a Java 17 JDK along with criteria
+    to decide if the timeline should be delayed
+
+Non-Goals:
+
+*   Providing a rationale for upgrading Java; all active Java projects need to
+    update their JDK version eventually as libraries we depend upon will also
+    depend on new versions of the JDK
+*   Debating which Java version we should upgrade to; we take it as a given that
+    only LTS releases are acceptable for most organizations, especially for
+    running critical infrastructure like Spinnaker
+*   Documenting and solving technical issues that present themselves in the
+    migration; those will be handled separately, but this document concerns
+    itself primarily with the larger processes and timeline
+*   Detailing the required release system changes; since the releases are still
+    entirely maintained by Google, these changes aren't particularly relevant to
+    the wider community
```

Comment:
```suggestion
*   Detailing the required release system changes; since the releases have recently been
     significantly refactored, the specifics of the process should be straightforward and consistent
     across repositories.
```